requests/util.js

```diff
@@ -0,0 +1,63 @@
+function createCustomElement(domObjectMap) {
+  const el = document.createElement(domObjectMap.tagName);
+  for (const [key, value] of Object.entries(domObjectMap)) {
+    if (key === 'tagName') {
+      continue;
+    }
+    if (key === 'eventListeners') {
+      value.forEach((obj) => {
+        el.addEventListener(obj.event, obj.func);
+      });
+    }
+    if (key === 'class') {
+      if (Array.isArray(value)) {
+        el.classList.add(...value);
+      } else {
+        el.classList.add(value);
+      }
+    } else if (key === 'child') {
+      el.append(...value);
+    } else {
+      el[key] = value;
+    }
+  }
+  return el;
+}
+
+function getOooQueryParamsString(query) {
+  let queryParam = 'type=OOO';
+  if (
+    query.state !== undefined &&
+    query.state !== null &&
+    query.state !== 'ALL'
+  ) {
+    queryParam += `&state=${query.state}`;
+  }
+  return '?' + queryParam.toString();
+}
+
+function convertDateToReadableStringDate(date) {
+  if (date !== undefined && date !== null) {
+    return (
+      new Date(date).getDate() +
+      ' ' +
+      new Date(date).toLocaleString('default', { month: 'short' }) +
+      ' ' +
+      new Date(date).getFullYear()
+    );
+  }
+  return 'NA';
+}
```

Comment:
`convertDateToReadableStringDate`  function converts a date into a human-readable string format.

Handles undefined or null dates gracefully and returns `N/A` in such cases.
  
While functional, consider optimizing for reusability by allowing the format to be customizable (e.g., accepting a format string parameter).

Reply:
this edge case is handled. please check.